src/devices/Common/Iot/Device/Common/IWritablePinSegment.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Interface that abstracts pin multiplexing.
+    /// </summary>
+    public interface IWritablePinSegment
+    {
+        /// <summary>
+        /// Length of segment (number of pins)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Writes a high or low value to pin
+        /// </summary>
+        void Write(int pin, int value);
```

Comment:
I suggest to use PinValue instead of `int`. Also I think this abstraction is very similar to already existing GpioController - I suspect if you added something like MappingGpioController/MappingGpioDriver you could achieve similar effect

Reply:
I thought of that, and it may be the right option. My reason for not choosing it is that `PinValue` seems very GPIO-oriented, which this is not.

Reply:
I'm still thinking about this. Maybe you are right. Maybe `PinValue` can be considered generic. I can take a run at making that change.

Reply:
This is the problem ... I have to cast `PinValue` to `int` here to enable that: https://github.com/richlander/iot/blob/Sn74hc595/src/devices/Sn74hc595/Sn74hc595.cs#L115

Thoughts?

Reply:
I could switch the `Shift` method to `PinValue` and then leave [ShiftByte](https://github.com/richlander/iot/blob/Sn74hc595/src/devices/Sn74hc595/Sn74hc595.cs#L138) as `byte`. Thoughts?

Reply:
Done.